sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -287,26 +259,29 @@ public synchronized void onTransactionStart(@NotNull ITransaction transaction) {
 
     // cpu max frequencies are read with a lambda because reading files is involved, so it will be
     // done in the background when the trace file is read
-    return new ProfilingTraceData(
-        traceFile,
-        transactionList,
-        transaction,
-        Long.toString(transactionDurationNanos),
-        buildInfoProvider.getSdkInfoVersion(),
-        abis != null && abis.length > 0 ? abis[0] : "",
-        () -> CpuInfoUtils.getInstance().readMaxFrequencies(),
-        buildInfoProvider.getManufacturer(),
-        buildInfoProvider.getModel(),
-        buildInfoProvider.getVersionRelease(),
-        buildInfoProvider.isEmulator(),
-        totalMem,
-        options.getProguardUuid(),
-        versionName,
-        versionCode,
-        options.getEnvironment(),
-        isTimeout
-            ? ProfilingTraceData.TRUNCATION_REASON_TIMEOUT
-            : ProfilingTraceData.TRUNCATION_REASON_NORMAL);
+    ProfilingTraceData profilingTraceData =
+        new ProfilingTraceData(
+            traceFile,
+            transactionList,
+            transaction,
+            Long.toString(transactionDurationNanos),
+            buildInfoProvider.getSdkInfoVersion(),
+            abis != null && abis.length > 0 ? abis[0] : "",
+            () -> CpuInfoUtils.getInstance().readMaxFrequencies(),
+            buildInfoProvider.getManufacturer(),
+            buildInfoProvider.getModel(),
+            buildInfoProvider.getVersionRelease(),
+            buildInfoProvider.isEmulator(),
+            totalMem,
+            options.getProguardUuid(),
+            versionName,
+            versionCode,
+            options.getEnvironment(),
+            isTimeout
+                ? ProfilingTraceData.TRUNCATION_REASON_TIMEOUT
+                : ProfilingTraceData.TRUNCATION_REASON_NORMAL);
+
+    HubAdapter.getInstance().captureProfile(profilingTraceData);
```

Comment:
How about we move this into the constructor? Then we should be also able to test it properly without overriding `Sentry.setCurrentHub`

Perhaps, we could even make `AndroidTransactionProfiler` an integration, but I'm not sure if it was ever discussed before @marandaneto

Reply:
Maybe create an envelope out of the `profilingTraceData` and call `captureEnvelope` instead?
Your call here @romtsn 
I don't recall any discussion related to `AndroidTransactionProfiler` as an integration.

Reply:
Out of curiosity, why is it preferred to create an Envelope and call `captureEnvelope` instead of calling `captureProfile`/`captureTransaction`/`captureWhatever`?
Is it for consistency or are there other reasons? @marandaneto 
I can quickly create the envelope and drop the `captureProfile` methods, if it's better

Reply:
Because this expands the public API and is not needed.

Reply:
@adinauer  Actually, `HubAdapter` is public, and it has a `captureProfile` method

Reply:
`HubAdapter` is just to make Sentry static class testable.

Reply:
That's fine, but `HubAdapter` isn't marked as internal, so technically there's nothing preventing users to actually use it.
Just wanted to make sure we are aware of this

Reply:
@marandaneto i made the change. Would you take another look before merging?